### PR TITLE
add Simple transistor

### LINKED ISSUE #104: add Simple transistor
### Description:

- add source_simple_transistor

### PR DESCRIPTION
**Pull request by Anshgrover23**
### Description:

- add source_simple_transistor

Fixes #104

### CODE REVIEW

**Comment by Anshgrover23** [comment]
@seveibar please check this.

File: src/source/source_simple_transistor.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_transistor = source_component_base.extend({
+  ftype: z.literal("simple_transistor"),
+  transistorType: z.enum(["NPN", "PNP"]),
+  pinLabels: z.object({
+    emitter: z.string(),
+    base: z.string(),
+    collector: z.string(),
+  }),
```
Comment (imrishabh18) [inline src/source/source_simple_transistor.ts:16]: I don't think we will need pinLabels separately. We can get the labels from the `schematic_port`

**Review by imrishabh18 — CHANGES_REQUESTED** [review]

File: src/source/source_simple_transistor.ts (hunk 1 above)
Comment (Anshgrover23) [reply src/source/source_simple_transistor.ts]: done.

**Comment by imrishabh18** [comment]
@Anshgrover23 I have requested some changes, please address those before asking for review again

File: src/source/source_simple_transistor.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import {
+  source_component_base,
+  type SourceComponentBase,
+} from "src/source/base/source_component_base"
+import { voltage, current } from "src/units"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_simple_transistor = source_component_base.extend({
+  ftype: z.literal("simple_transistor"),
+  transistorType: z.enum(["NPN", "PNP"]),
```
Comment (seveibar) [inline src/source/source_simple_transistor.ts:11]: this project uses snake case

File: src/source/source_simple_transistor.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  vce: voltage.optional(),
+  vbe: voltage.optional(),
+  ic: current.optional(),
```
Comment (seveibar) [inline src/source/source_simple_transistor.ts:14]: don't abbreviate, also i don't know what these are so once you unabbreviate there's a chance they're not needed (can't know unless they are fixed)

**Review by seveibar — CHANGES_REQUESTED** [review]

**Comment by Anshgrover23** [comment]
@seveibar @imrishabh18 done with all changes also rishabh bhaiya does this schematic port implementation we needed.

**Comment by seveibar** [comment]
this repo uses snake case between words, look at other source components none of them have schematic ports.

**Comment by Anshgrover23** [comment]
@seveibar yes i knew that , as according to that i have removed pinlabels but @imrishabh18 said this 
image
so do i have to remove all this.

**Review by seveibar — CHANGES_REQUESTED** [review]
see comment

**Review by seveibar — APPROVED** [review]

**Pull request merged**